docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,538 @@
+              The compatibility MUST be referenced by `id`.
+
+            - **`oneOf`** *array of strings*
+
+              This OPTIONAL property references a list of target compatibility nodes defined in the `spec.compatibilities` with a condition that at least one node has to pass that container is considered to be compatible with a host.
+              The compatibility MUST be referenced by `id`.
+
+            - **`noneOf`** *array of strings*
+
+              This OPTIONAL property references a list of target compatibility nodes defined in the `spec.compatibilities` with a condition that any node on the list cannot pass that container is considered to be compatible with a host.
+              The compatibility MUST be referenced by `id`.
+
+      - **`validationCriteria`** *array of objects*
+
+        This REQUIRED property defines compatibility criteria.
+
+        - **`graphs`** *array of strings*
+
+          This REQUIRED property references graphs names from `spec.relations.graphs`. The graphs are used to validate compatibility.
+
+        - **`condition`** *string*
+
+          This REQUIRED property specify the compatibility validation rule.
+          Allowed values: _all_ - all referenced graphs must pass; _one_ - one referenced graph must pass from the list.
+
+        - **`annotations`** *string-string map*
+
+          This OPTIONAL property describes user defined annotations.
+
+Example of use:
+
+- Simple requirements - container supporting only NVIDIA GPUs
+
+```json
+{
+  "spec": {
+    "compatibilities": [
+      {
+        "id": "nvidiaGPU",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.pci.class-id": "0380",
+          "hardware.pci.vendor-id": "10de",
+          "kernel.configuration": [
+            "CONFIG_HOTPLUG_PCI_PCIE=y",
+            "CONFIG_MODULES=y",
+            "CONFIG_MODULE_UNLOAD=y",
+            "CONFIG_PCI_MMCONFIG=m",
+            "CONFIG_DRM_NOUVEAU=n"
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+- Complex requirements with relations - container supporting multiple GPUs
+
+```json
+{
+  "spec": {
+    "compatibilities": [
+      {
+        "id": "intelCpu",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.cpu.vendor": "GenuineIntel",
+          "hardware.cpu.virtualization": "VT-x",
+          "kernel.cmdline": ["intel_iommu=on"]
+        }
+      },
+      {
+        "id": "amdCpu",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.cpu.vendor": "AuthenticAMD",
+          "hardware.cpu.virtualization": "AMD-V",
+          "kernel.cmdline": ["amd_iommu=pt"]
+        }
+      },
+      {
+        "id": "vfio",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "kernel.modules": [
```

Comment:
This is even harder - now it's not just a key -> interface value,  but the interface is another dictionary. This makes me very anxious from a developer perspective.

Reply:
I know interfaces are not ideal. However, I think we should value more the user experience. If forcing users to flatten composite types will be a pain then we will have to find some compromise.

Anyway, as mentioned here https://github.com/opencontainers/wg-image-compatibility/pull/6#discussion_r1487407555 I have to try to build such spec.

Reply:
Answered here https://github.com/opencontainers/wg-image-compatibility/pull/6#discussion_r1497147795